StencilFrontend.md

```diff
@@ -0,0 +1,353 @@
+<a href= "http://swift-at-ibm-slack.mybluemix.net/">
+<img src="http://swift-at-ibm-slack.mybluemix.net/badge.svg"  alt="Slack">
+</a>
+</p>
+
+Now that there is a FoodServer backend for the [FoodTracker](https://github.com/IBM/FoodTrackerBackend) app, you can add a website front end so that the page displayed by your server is more visually appealing than returning a JSON array of meals. The steps below demonstrate how to add a Stencil template to your server so that it can return dynamic web pages, using [Kitura-TemplateEngine](https://github.com/IBM-Swift/Kitura-TemplateEngine) and [Kitura-StencilTemplateEngine](https://github.com/IBM-Swift/Kitura-StencilTemplateEngine).
+
+
+## Pre-Requisites:
+These instructions follow on from the FoodTracker application and server created by following the [FoodTrackerBackend](https://github.com/IBM/FoodTrackerBackend) tutorial. If you have completed the FoodTracker Backend there are no further pre-requisites.
+
+If you have not completed the [FoodTrackerBackend](https://github.com/IBM/FoodTrackerBackend) tutorial go to the [CompletedFoodTracker](https://github.com/IBM/FoodTrackerBackend/tree/CompletedFoodTracker) branch and follow the [README](https://github.com/IBM/FoodTrackerBackend/blob/CompletedFoodTracker/README.md) instructions.
+
+## Using A Stencil Template
+The FoodTracker application is taken from the Apple tutorial for building your first iOS application. In the [FoodTrackerBackend](https://github.com/IBM/FoodTrackerBackend) tutorial, we created a server and connected it to the iOS application. This means that when meals are added they are posted to the server, a user can then view these meals on [localhost:8080/meals](http://localhost:8080/meals). When you perform a HTTP `GET` request to the server, you are returned a JSON array which contains your stored meals. To present this data in a more appealing way we will use [Stencil](https://github.com/kylef/Stencil). Stencil is a template language for Swift which allows you to embed code into a HTML document to create a dynamic web page.
+
+### Create the Stencil File
+To use templates with a Kitura router, you must create a `.stencil` template file which describes how to embed code into the returned HTML. By default, the Kitura router gets the template files from the `./Views/` directory in the directory where Kitura runs.
+
+1. In the terminal, change into the FoodServer directory:
+```
+cd ~/FoodTrackerBackend/FoodServer
+```
+2. Create the Views folder and change into it:
+```
+mkdir Views
+cd Views
+```
+3. Create a `FoodTemplate.stencil` file.
+We will use terminal commands for file creation but you can use a text editor of your choice.
+```
+cat > FoodTemplate.stencil
+```
+**Note:** You can exit `cat` using `CTRL+D`
+
+4. Add the following Stencil template code to display the meals:
+
+```
+<html>
+There are {{ meals.count }} meals. <br />
+
+{% for meal in meals %}
+    - {{ meal.name }} with rating {{ meal.rating }}. <br />
+    <br />
```

Comment:
Please can we remove this line break (the one on line 51).

It would be nice to see this as follows which is more like the Stencil example it is based on (I don't think we need the full stop after the meal name and ratings as they are only list items):

```
There are 2 meals.

 - Pasta bolognaise with rating 2
 - Spaghetti carbonara with rating 5
```